### PR TITLE
Migrating to Lego 2.0.0

### PR DESCRIPTION
**Pull request by shrey-garg**

### CODE REVIEW

File: api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java [hunk 1]
```diff
@@ -24,7 +24,7 @@
 import java.util.Collection;
 import java.util.List;
 
-public class ListDataType<T> extends ArrayList<T> implements DataType {
+public class ListDataType<T extends DataType> extends ArrayList<T> implements DataType {
```
Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: This can't be enforced. Not sure why this change now?

Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: Shouldnt we disallow sending anything in a list that is not of type DataType?

Comment (pmohankumar) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: You may want to look at OrchestratorDataSource.java. ListDataType and MapDataType are used to convert response coming out of Hydra's composer to a DataType (to maintain DataSource contract of returning DataType). ListDataType is not used to convey a list of datatypes :)

Comment (shrey-garg) [inline api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java]: Reverted this

File: container/src/main/java/com/flipkart/poseidon/Poseidon.java [hunk 2]
```diff
@@ -149,14 +149,17 @@ private Handler getRewriteHandler(Handler handler) {
...
+            if (rewriteFilePath != null && !(rewriteFilePath = rewriteFilePath.trim()).isEmpty()) {
+                JavaType listRuleType = getMapper().getTypeFactory().constructParametricType(List.class, RewriteRule.class);
+                List<RewriteRule> rules = getMapper().readValue(new FileInputStream(rewriteFilePath), listRuleType);
+                for (RewriteRule rule : rules) {
+                    if (rule.isActive()) {
+                        RewriteRegexRule regexRule = new RewriteRegexRule();
+                        regexRule.setRegex(rule.getFrom());
+                        regexRule.setReplacement(rule.getTo());
+                        rewriteHandler.addRule(regexRule);
+                    }
```
Comment (pmohankumar) [inline container/src/main/java/com/flipkart/poseidon/Poseidon.java]: Shrey, can you test this before I merge this PR? I mean from your app don't supply rewriteFilePath and make sure APIs work without any exceptions (mainly because we are still creating RewriteHandler). Another question is do we need to create a rewriteHandler when no rewrite file path isn't specified?

**Comment by pmohankumar** [comment]
Except for the comment around RewriteHandler, everything else looks good

File: container/src/main/java/com/flipkart/poseidon/Poseidon.java (hunk 2 above)
Comment (shrey-garg) [inline container/src/main/java/com/flipkart/poseidon/Poseidon.java]: I have now made the RewriteHandler optional and tested it both ways, when rewrite file path is present or absent. Works fine.
